utils/config.go

```diff
@@ -252,21 +273,32 @@ func NewConfig(ctx *cli.Context, mode ArgumentMode) (*Config, error) {
 	cfg := createConfigFromFlags(ctx)
 
 	// create config context for sharing common arguments
-	cc := NewConfigContext(cfg)
+	cc := NewConfigContext(cfg, ctx)
 
 	// check if chainID is set correctly
 	err = cc.setChainId()
 	if err != nil {
 		return nil, fmt.Errorf("cannot get chain id; %v", err)
 	}
 
+	// set chain config
+	chainConfig, err := GetChainConfig(cc.cfg.ChainID)
+	if err != nil {
+		return nil, fmt.Errorf("cannot create chain config; %w", err)
+	}
+
+	cc.cfg.ChainCfg = chainConfig
```

Comment:
nit: save a few lines
```suggestion
	cc.cfg.ChainCfg, err := GetChainConfig(cc.cfg.ChainID)
	if err != nil {
		return nil, fmt.Errorf("cannot create chain config; %w", err)
	}
```